griptape/drivers/prompt/routellm_prompt_driver.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from attrs import Factory, define, field
+
+from griptape.drivers import BasePromptDriver
+from griptape.tokenizers.dummy_tokenizer import DummyTokenizer
+from griptape.utils.import_utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from routellm.controller import Controller
+
+    from griptape.common import (
+        DeltaMessage,
+        Message,
+        PromptStack,
+    )
+    from griptape.tokenizers import BaseTokenizer
+
+
+@define
+class RouteLlmPromptDriver(BasePromptDriver):
+    """[RouteLlm](https://github.com/lm-sys/RouteLLM) Prompt Driver.
+
+    Attributes:
+        strong_prompt_driver: Prompt Driver to use when routing to the strong model.
+        weak_prompt_driver: Prompt Driver to use when routing to the weak model.
+        threshold: Cost threshold when routing between models.
+        router: Router to use, defaults to "mf".
+        client: RouteLlm Controller.
+        tokenizer: Tokenizer to use, defaults to DummyTokenizer. After running, it will be set to the tokenizer of the routed prompt driver.
+    """
+
+    model: str = field(init=False, default=None)
+    strong_prompt_driver: BasePromptDriver = field(
+        kw_only=True,
+        metadata={"serializable": False},
+    )
+    weak_prompt_driver: BasePromptDriver = field(
+        kw_only=True,
+        metadata={"serializable": False},
+    )
+    router: str = field(kw_only=True, default="mf", metadata={"serializable": True})
+    threshold: float = field(kw_only=True, metadata={"serializable": True})
+    client: Controller = field(
```

Comment:
I think you should name this field `controller` instead of `client`. When I read `client` I thought this thing was actually calling a service, but its not, its all local.